alacritty_terminal/src/term/search.rs

```diff
@@ -290,7 +285,16 @@ impl<T> Term<T> {
     ) {
         match direction {
             Direction::Right if cell.flags.contains(Flags::WIDE_CHAR) => {
-                iter.next();
+                // If advances `iter` when finds a wide character without wide character spacer,
+                // one character is skipped unintentionally. So advances `iter` only if next
+                // item(cell) exists and it is wide character spacer.
+                let idxd = match iter.peek() {
+                    Some(idxd) => idxd,
+                    None => return
+                };
+                if idxd.cell.flags.contains(Flags::WIDE_CHAR_SPACER) {
+                    iter.next();
+                }
```

Comment:
I think you might be overcomplicating things, there's no need to do this peeking to determine if there's going to be a spacer or not, all we need to make sure of is that there's a cell after the wide char. So we just need to check the current cell's column.

Reply:
Oh, sure!